### DISCUSSION TITLE
Can't parse output from defaults domains command in macOS

### DISCUSSION THREAD

**Discussion by thesobercoder** [johnkerl/miller#1599] [Q&A] [answered]
Dumb question, but why does this command return nothing, am I doing something wrong?

```sh
defaults domains | mlr --csv --opprint cat
```

I've verified tha `awk` works with the csv output as expected.

```sh
defaults domains | awk -F, '{for(i=1; i<=NF; i++) print $i}'
```

**Comment by aborruso** [comment]
Could you share a sample raw output of the `defaults domains ` command?

**Comment by johnkerl** [comment, marked as answer]
@thesobercoder the output of `defaults domains` is indeed comma-separated, but, with only one line. In CSV this means the header line, and no data. That in turn means zero records.

There is #314 open to address this.

However, I suspect that CSV-with-header is not what you want here.

You might use `mlr --csv --headerless-input` (keystroke-saver: `mlr -c --hi`)

Some examples:

```
$ defaults domains > dd.txt

$ mlr --icsv --hi --oxtab --from dd.txt clean-whitespace | head
1   .GlobalPreferences_m
2   1812769364
3   Avatar Cache Index
4   Cisco-Systems.Spark
5   ContextStoreAgent
6   LighthouseBitacoraFramework
7   MiniLauncher
8   MobileMeAccounts
9   ScopedBookmarkAgent
10  ZoomChat

$ mlr --inidx --ifs comma --ojson --from dd.txt clean-whitespace | head
[
{
  "1": ".GlobalPreferences_m",
  "2": 1812769364,
  "3": "Avatar Cache Index",
  "4": "Cisco-Systems.Spark",
  "5": "ContextStoreAgent",
  "6": "LighthouseBitacoraFramework",
  "7": "MiniLauncher",
  "8": "MobileMeAccounts",
```
